src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -34,17 +35,35 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
 
         public virtual async Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
         {
-            var newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            bindingContext.ModelName);
-
+            var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
             var modelBindingResult = await TryBind(newBindingContext);
-            if (modelBindingResult == null && !string.IsNullOrEmpty(bindingContext.ModelName)
-                && bindingContext.FallbackToEmptyPrefix)
+
+            if (bindingContext.FallbackToEmptyPrefix && !string.IsNullOrEmpty(bindingContext.ModelName))
             {
-                // fallback to empty prefix?
-                newBindingContext = CreateNewBindingContext(bindingContext,
-                                                            modelName: string.Empty);
-                modelBindingResult = await TryBind(newBindingContext);
+                var fallback = modelBindingResult == null;
+                if (!fallback && modelBindingResult.IsModelSet && modelBindingResult.IsEmptyModel)
+                {
+                    // Special case the empty object (i.e. contains at most default values) MutableObjectModelBinder
+                    // may return.
+                    fallback = true;
+
+                    // Remove any validation errors from first attempt; these are likely Required errors.
+                    // Clear entries completely, unlike ModelState.ClearValidationState().
```

Comment:
`ClearValidationState` doesn't remove the entries, it just clears the contained `ModelState.Entries` entries and removes `ModelState.ValidationState` references.  this leaves the `ModelState.Value` untouched.  that isn't correct for fallback scenarios.

I mentioned in the PR description that unit and functional tests are to come.  the bugs are:
- #1865 (still needs a reword): `MutableObjectModelBinder` returns a non-`null` result containing a value whenever it can create that value.  this looks to the calling `CompositeModelBinder` like success and prevents fallback to the empty prefix when no properties in a top-level model are actually set.
- #2129: `GenericModelBinder` returns a non-`null` result whenever it can create a binder for the given type.  this also prevents fallback to the empty prefix when no provider has data.

this particular code is part of the #1865 fix.